app/controllers/datasets_controller.rb

```diff
@@ -15,13 +15,24 @@ def import_ona
   end
 
   def upload
-    uploaded_io = params["file"]
-    filename_to_upload = uploaded_io.original_filename
+    validate_upload_params
+    filename = (params["file"] && params["file"].original_filename) || params["name"]
+    FileUtils.mkdir_p DatasetsChannel.directory_for(filename)
 
-    FileUtils.mkdir_p DatasetsChannel.directory_for(filename_to_upload)
-
-    File.open(DatasetsChannel.path_for(filename_to_upload), 'wb') do |file|
-      file.write(uploaded_io.read)
+    if params["url"] 
+      sheetId = sheet_id_match(params["url"])
+      data_enum = SpreadsheetService.get_data(sheetId)
+      CSV.open(DatasetsChannel.path_for(filename), 'wb') do |file|
```

Comment:
To avoid dealing with special characters as you mentioned before, I'm writing the file in a `CSV.open` context. @bcardiff , do you think is enough to tackle that problem?

Reply:
👍 on the CSV.open
👎 on using an action controller to run this procedure that could timeout if the spreadsheet is long. Has this been tested with long datasets?

Reply:
We tested it on a 1MB file and it worked. I'm wondering how can I queue it in background. Can I call something like `run_process` passing rows as arguments?

Reply:
Ohhh, I think I've got it. I have to do the whole thing on background. The argument I have to pass to the process is the `sheetId`. It's analogous to `import-dataset` calling `Indexing.rb`